### PR TITLE
Add chapter 19.

### PR DESCRIPTION
**Pull request by OrangeToucan**

### CODE REVIEW

File: docs/19-compiled-programs.markdown [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+`RSUBR`s can call any `APPLICABLE` object, all in a uniform manner. In
+general, a call to an F/SUBR is linked up at assembly/compile time so
+that the calling instruction (UUO) points directly at the code in the
+interpreter for the F/SUBR. However, the locations of most other
+`APPLICABLE`s are not known at assembly/compile time. Therefore, the
+calling UUO is set up to point at a slot in the reference vector (by
+indexing off accumulator `R`). This slot initially contains the `ATOM`
+whose G/LVAL is the called object. The calling mechanism (UUO handler)
+causes control to be transferred to the called object and, depending
+on the state of the `RSUBR-LINK` flag, the `ATOM` will be replaced by
```
Comment (taradinoc) [inline docs/19-compiled-programs.markdown]: `RSUBR`-link

File: docs/19-compiled-programs.markdown [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+its G/LVAL. (If the call is of the "quick" variety, the called `RSUBR`
+or `RSUBR-ENTRY` will be `CHTYPE`d to a `QUICK-RSUBR` or
+`QUICK-ENTRY`, respectively, before replacement.) Regardless of the
+`RSUBR-LINK` flag's state, calls to `FUNCTION`s are never permanently
```
Comment (taradinoc) [inline docs/19-compiled-programs.markdown]: `RSUBR`-link

File: docs/19-compiled-programs.markdown [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+code, regardless of the current G/LVAL of the called `ATOM`. Thus,
+while testing `RSUBR`s, you may want to disable linking, by calling
+the `RSUBR-LINK` `SUBR` with a `FALSE` argument. Calling it with a
+non-`FALSE` argument enables linking thereafter. It returns the
+previous state of the link flag, either `T` or `#FALSE ()`. Calling it
+with no argument returns the current state.
+
+## 19.4. Pure and Impure Code
+
+The first element of an `RSUBR` is the code vector. of `TYPE` `CODE`
```
Comment (taradinoc) [inline docs/19-compiled-programs.markdown]: period -> comma

File: docs/19-compiled-programs.markdown [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+`SUBR`, which generates a pure code vector.) Pure `RSUBR`s may also
+move around, but only by being included in MDL's page map at different
+places. Once again `M` can be used exactly as before to do
+location-independent address referencing. Individual pure code vectors
+can be "unmapped" (marked as being not in primary storage but in their
+original pure-code disk files) if the space in storage allocated for
+pure code is exhausted. An unmapped `RSUBR` is mapped in again
+whenever needed. All pure `RSUBR`s are unmapped before a `SAVE` file
+is written, so that the code is not duplicated on disk. A purified
+`RSUBR` must use `RGLOC` "(relative GLOC") instead of `GLOC`. `RGLOC`
```
Comment (taradinoc) [inline docs/19-compiled-programs.markdown]: move `"` inside parentheses before "relative GLOC"

File: docs/19-compiled-programs.markdown [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+produces objects of `TYPE` `LOCR` instead of `LOCD`.
+
+# 19.5. TYPE -C and TYPE -W
```
Comment (taradinoc) [inline docs/19-compiled-programs.markdown]: no space in `TYPE-C` or `TYPE-W`

File: docs/19-compiled-programs.markdown [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+`#CODE` followed by a `UVECTOR` of `PRIMTYPE` `WORD`s. `BINARY` files
+are generally slow to load, because of all the parsing that must be
+done.
+
+An `NBIN` file contains a mixture of ASCII characters and binary code.
+The start of a binary portion is signalled to `READ` by the character
+control-C, so naive readers of an `NBIN` file under ITS may
+incorrectly assume that it ends before any binary code appears. An
+`NBIN` file cannot be edited with a text editor. An `RSUBR` is written
+in `NBIN` format by being `PRINT`ed on a "`PRINTB`" `CHANNEL`. The
```
Comment (taradinoc) [inline docs/19-compiled-programs.markdown]: quote marks should be inside the backticks in `"PRINTB"`

**Comment by OrangeToucan** [comment]
Done!

**Comment by OrangeToucan** [comment]
Did I miss something?

**Comment by taradinoc** [comment]
Nope, I just forgot to merge this. Thanks for reminding me.

**Pull request merged**